src/vs/workbench/browser/dnd.ts

```diff
@@ -176,14 +178,25 @@ export class ResourcesDropHandler {
 	}
 
 	async handleDrop(event: DragEvent, resolveTargetGroup: () => IEditorGroup | undefined, afterDrop: (targetGroup: IEditorGroup | undefined) => void, targetIndex?: number): Promise<void> {
-		const untitledOrFileResources = extractResources(event).filter(r => this.fileService.canHandleResource(r.resource) || r.resource.scheme === Schemas.untitled);
+
```

Comment:
Please put this logic in the `drop` function of `dragAndDropController.ts` instead. The controller class has a good API for any tree that needs custom implementation, and you should be able to do this without changing the underlying logic of the drag and drop.